worker/uploader.go

```diff
@@ -116,14 +116,37 @@ outer:
 			}
 
 			// execute it
-			elapsed, err := u.execute(req)
-
-			// the uploader's contract got renewed, requeue the request
-			if errors.Is(err, errMaxRevisionReached) {
-				if u.tryRefresh(req.sector.ctx) {
+			start := time.Now()
+			duration, err := u.execute(req)
```

Comment:
Hm re-reading my own PR I'm wondering whether it would have been better to extract all of this logic into a method on the uploader called `trackUpload(sectorUploadDuration, totalDuration time.Duration, overdrive bool, err error)` and write a unit test for it. That would mean we're not using the worker tester though but instead write a test where we're passing errors we construct ourselves. Wdyt @ChrisSchinnerl ?

Reply:
How about a function instead of a method which can be unit-tested additionally?

e.g. 
```go
if errors.Is(err, errMaxRevisionReached) {
  // requeue
} else {
  d, reason, success := myUnitTestableFunction(...)
  u.trackSectorUpload(success, d)
  if reason != "" {
    u.logger.Debugw("something about punishing the worker")
  }
}
```

Not covering `errMaxRevisionReached` is probably fine in that case

Reply:
Hm yeah ok I'll change it but it's a little more tricky because we don't always want to update consecutive failures.